app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java

```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
+    }
+
+    @Override
+    public void onConnectionSuspended(int i) {
 
-        Bitmap icon = BitmapFactory.decodeResource(getResources(),
-                R.drawable.ic_darren);
+    }
 
-        Bitmap croppedIcon = getCroppedBitmap(icon);
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
 
-        mCurrentMarker = mMap.addMarker(new MarkerOptions().position(latLng).title("Current Location").icon(BitmapDescriptorFactory.fromBitmap(croppedIcon)));
-      }
-    } catch (SecurityException se) {
-      lawg.e("se: " + se);
     }
-  }
 
-  @Override
-  public void onConnectionSuspended(int i) {
+    public Bitmap getCroppedBitmap(Bitmap bitmap) {
+        Bitmap output = Bitmap.createBitmap(bitmap.getWidth(),
+                bitmap.getHeight(), Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(output);
+
+        final int color = 0xff424242; //Changing this to R.color.transparent removes icon
 
-  }
 
-  @Override
-  public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+        final Paint paint = new Paint();
+        final Rect rect = new Rect(0, 0, bitmap.getWidth(), bitmap.getHeight());
 
-  }
+        paint.setAntiAlias(true);
+        canvas.drawARGB(0, 0, 0, 0);
+        paint.setColor(color);
 
-  public Bitmap getCroppedBitmap(Bitmap bitmap) {
-    Bitmap output = Bitmap.createBitmap(bitmap.getWidth(),
-            bitmap.getHeight(), Bitmap.Config.ARGB_8888);
-    Canvas canvas = new Canvas(output);
+        canvas.drawCircle(bitmap.getWidth() / 2, bitmap.getHeight() / 2,
+                bitmap.getWidth() / 2, paint);
+        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
+        canvas.drawBitmap(bitmap, rect, rect, paint);
+
+        return output;
+    }
 
-    final int color = getResources().getColor(R.color.transparent);
-    final Paint paint = new Paint();
-    final Rect rect = new Rect(0, 0, bitmap.getWidth(), bitmap.getHeight());
 
-    paint.setAntiAlias(true);
-    canvas.drawARGB(0, 0, 0, 0);
-    paint.setColor(color);
+    //Toasts the tapped points coords
+    @Override
+    public void onMapClick(LatLng point) {
+        Toast.makeText(this, "Tapped point is: " + point, Toast.LENGTH_LONG).show();
+        VisibleRegion vRegion = mMap.getProjection().getVisibleRegion();
+        LatLng upperLeft = vRegion.farLeft;
+        LatLng lowerRight = vRegion.nearRight;
+        //Logs the visible area of the map
+        Log.d("Coord", "Top left = " + upperLeft + " and Bottom right = " + lowerRight);
 
-    canvas.drawCircle(bitmap.getWidth() / 2, bitmap.getHeight() / 2,
-            bitmap.getWidth() / 2, paint);
-    paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
-    canvas.drawBitmap(bitmap, rect, rect, paint);
+    }
 
-    return output;
-  }
+    @Override
+    public void onMapLongClick(LatLng point) {
+        Toast.makeText(this, "Held point is: " + point, Toast.LENGTH_LONG).show();
```

Comment:
Actually this longclick one isn't necessary since you already got it in the method above